### ISSUE TITLE
SPIKE: test tailscale with team

### ISSUE THREAD

**Issue by bloodearnest** [opensafely-core/sysadmin#82]
We need to test that tailscale will provide what we need with the whole team

Plan is to set it up on the test backend

**Comment by bloodearnest** [comment]
tl;dr: we'd need to pay, which delays using things right now.

## pros
 - ephemeral keys
    - no key management
    - goodbye hacky kissh
 - simple automatic dns
 - port 22 not on public internet
 - github/google auth w/2fa
 - makes internal services easy
 - better ssh audit log (managed by tailscale, more tamper proof, can do session logging)
 - easy access to dev machines
   - e.g. "can you have a look at http://simon:8000 and check my local wip?"
   - dev willing, can also enable ssh 
     - e.g. to jon's sqlserver lab setup
     - e.g. my lxc to run backend server tests for those on macs

## cons:
 - cannot easily lock down our existing orgs due to ACL limits
    - any org user could ssh in
    - very limited group sizes on free plan (2 unique users!)
       - Team plan is only 5 unique users!
       - We would need business plan ($7.50/user/month)
    - cannot currently use org teams, need to redefine teams in tailscale
 - org members can ssh in any local user except root
   - can lock down, but requires ACL users as above
 - outsourcing key component

 
## questions:

### Shared login account?

A big part of the of point of per-user accounts on servers was ssh key management. We could instead move to a model where we all ssh into a shared account. e.g. `bennett@dokkuN`. This has the advantage of simpler account management, and shared bash history. Tailscale still maintains an audit of who logged in. This would mean either passwordless sudo or a shared sudo password. 

However there are other benefits of separate accounts. e.g. per user sudo passwords, scratch area to stage things.

### Are developers happy to expose their machines on the network?

Team will only be able to access ports if you bind to 0.0.0.0 or tailscale ip directly.

ssh only available if you activate it, but would need ACLs to stop them logging in as you.

**Comment by bloodearnest** [comment]
By default, ssh is not enabled on your personal device.

When you do enable it, by default, only you can ssh into in.

We'd have to add ACLs to allow others to ssh in.

**Comment by sebbacon** [comment]
While it will be great to have the option on an opt-in basis, I am strongly of the opinion `0.0.0.0` should be locked down by default, on the principle of least surprise.

**Comment by ghickman** [comment]
For Django at least you'd have to opt-in to bind to 0.0.0.0, but having that choice available for demos would be _chefs kiss_

**Comment by bloodearnest** [comment]
> While it will be great to have the option on an opt-in basis, I am strongly of the opinion `0.0.0.0` should be locked down by default, on the principle of least surprise.
> 
> edit: Although "if you bind to 0.0.0.0 or tailscale ip directly" makes it sound like this _may_ be opt-in by default?

It depends how you run it locally. If you run on 127.0.0.1 (the default for django and most dev setups), it won't be available to the tailnet.

Additionally, we can control what ports are accessible are routable via ACLs

So, we could have an ACL that allows port 8000 access to all machines. If you run something something that listens on 0.0.0.0:8000 (or your_tailscale_ip:8000), then other users on the tailnet will be able to access it.

The ACLs allow us to decide what ports are routable. The owner of the machine decides what services are running on what ips/ports.

**Comment by bloodearnest** [comment]
https://github.com/ebmdatalab/sysadmin/pull/233
